tesseract/tesseract_scene_graph/include/tesseract_scene_graph/resource_locator.h

```diff
@@ -117,9 +119,12 @@ class SimpleLocatedResource : public tesseract_common::Resource
 
   std::shared_ptr<std::istream> getResourceContentStream() override;
 
+  tesseract_common::Resource::Ptr locateSubResource(const std::string& relative_path) override;
+
 protected:
   std::string url_;
   std::string filename_;
+  std::weak_ptr<SimpleResourceLocator> parent_;
```

Comment:
If you have two shared_ptr objects that point to each other, they will never be deleted because they created a mutual reference to each other. This is called a "reference loop" or "circular reference". weak_ptr is a weak reference, that stores a reference to the object, but does not increase the reference count. If the reference count goes to zero, weak_ptr will return nullptr. I am using it here because there is potentially a circular reference between the resource locator and the located resource.

Reply:
Ah, good to know. Thank you for the information.